src/main/java/com/github/manolo8/darkbot/core/api/GameAPIImpl.java

```diff
@@ -279,6 +283,42 @@ public void readMemory(long address, byte[] buffer, int length) {
         memory.readBytes(address, buffer, length);
     }
 
+    @Override
+    public DataReader readData(long address, int length) throws RuntimeException {
+        if (length <= 0 || length > DataReader.MAX_CHUNK_SIZE)
+            throw new ArrayIndexOutOfBoundsException("Length is <= 0 or exceeds max chunk size: " + DataReader.MAX_CHUNK_SIZE);
+
+        for (int i = 0; i < dataReaders.length; i++) {
+            AbstractDataReader reader = dataReaders[i];
+            if (reader == null) reader = dataReaders[i] = createReader(i);
+
+            Boolean result = reader.read(address, length);
+
+            if (result == null) {
+                reader.reset(0);
+                return reader;
+            }
+
+            if (result) return reader;
+        }
+
+        throw new RuntimeException("Every reader is busy");
```

Comment:
```suggestion
        throw new RuntimeException("All DataReaders are in use. Some code is calling readData and not closing the resource!");
```
Be a bit more specific here, it's always good to give a bit more info